homeassistant/components/upnp/__init__.py

```diff
@@ -160,6 +164,8 @@ async def async_setup_entry(hass: HomeAssistantType, config_entry: ConfigEntry)
     hass.data[DOMAIN]["devices"][device.udn] = device
     hass.config_entries.async_update_entry(
         entry=config_entry,
+        unique_id=device.unique_id,
+        title=device.name,
         data={**config_entry.data, "udn": device.udn, "st": device.device_type},
```

Comment:
Why is this saved every time ? You should only update it if something is different. 

Also by changing the title, users can no longer rename them in the UI.